tests/smolyay_test.py

```diff
@@ -0,0 +1,159 @@
+import numpy
+from smolyay.smolyak import IndexGrid,generate_compositions
+
+@pytest.fixture
+def expected_points():
+    """grid_point_indexes for dimension=2,exactness=2,index_per_level=[1,2,2]"""
+    return numpy.array([[0,0],[0,1],[0,2],[0,3],[0,4],[1,0],
+        [1,1],[1,2],[2,0],[2,1],[2,2],[3,0],[4,0]], dtype=numpy.int32)
+
+@pytest.fixture
+def expected_indexes():
+    """level_indexes for dimension=2,exactness=2,index_per_level=[1,2,2]"""
+    return [[0],[1,2],[3,4]]
+
+def test_initial_dimension():
+    """test initialized class returns correct dimension"""
+    test_class = IndexGrid(1,2,[1,2,3])
+    assert test_class.dimension == 1
+
+def test_initial_exactness():
+    """test initialized class returns correct exactness"""
+    test_class = IndexGrid(1,2,[1,2,3])
+    assert test_class.exactness == 2
+
+def test_initial_index_by_level():
+    """test initialized class returns correct index by level"""
+    test_class = IndexGrid(1,2,[1,2,3])
+    assert numpy.array_equiv(test_class.index_per_level,[1,2,3])
+
+def test_level_indexes(expected_indexes):
+    """test level indexes"""
+    test_class = IndexGrid(2,2,[1,2,2])
+    level_indexes_recieved = test_class.level_indexes
+    
+    assert level_indexes_recieved == expected_indexes
+
+def test_grid_point_index(expected_points):
+    """test grid point index"""
+    test_class = IndexGrid(2,2,[1,2,2])
+    grid_points = sorted(test_class.grid_point_indexes,key = lambda x:x[0])
+
+    assert numpy.array_equal(grid_points,expected_points)
+
+def test_level_indexes_extra_level_per_index(expected_indexes):
+    """test level indexes when level_per_index has more information"""
+    test_class = IndexGrid(2,2,[1,2,2,4,8])
+    level_indexes_recieved = test_class.level_indexes
+
+    assert level_indexes_recieved == expected_indexes
+
+def test_grid_point_index_extra_level_per_index(expected_points):
+    """test grid point index when level_per_index has more information"""
+    test_class = IndexGrid(2,2,[1,2,2,4,8])
+    grid_points = sorted(test_class.grid_point_indexes,key = lambda x:x[0])
+
+    assert numpy.array_equal(grid_points,expected_points)
+
+def test_index_per_level_expand():
+    """test if properties change if more is added to level_per_index"""
+    test_class = IndexGrid(1,2,[1,2,3])
+    grid_point_1 = test_class.grid_point_indexes
+
+    test_class.index_per_level = [1,2,3,4,5]
+    grid_point_2 = test_class.grid_point_indexes
+    
+    assert numpy.array_equal(grid_point_1,grid_point_2)
+
+def test_increase_exactness(expected_points):
+    """test if grid points are correct if exactness is increased"""
+    test_class = IndexGrid(2,1,[1,2,2])
+    grid_points_1 = sorted(test_class.grid_point_indexes,key = lambda x:x[0])
+    test_class.exactness = 2
+    grid_points_2 = sorted(test_class.grid_point_indexes,key = lambda x:x[0])
+    assert ((not numpy.array_equiv(grid_points_1,grid_points_2))
+            and numpy.array_equiv(grid_points_2, expected_points))
```

Comment:
Perform each of these assertions separately
```suggestion
    assert not numpy.array_equiv(grid_points_1,grid_points_2)
    assert numpy.array_equiv(grid_points_2, expected_points)
```